engine/src/main/java/org/hibernate/search/engine/search/query/SearchResult.java

```diff
@@ -19,7 +19,7 @@
 	/**
 	 * @return The total number of matching entities, ignoring pagination settings.
 	 */
-	long totalHitCount();
+	Long totalHitCount();
```

Comment:
As discussed, this shouldn't be. I believe you only had to do that because you use `SearchResult` as a return even when fetching just hits, even though that interface only makes sense when we actually have a total hit count.

I can see two solutions:

1. Use a `long`, but make it default to `0` when we don't know. Dirty but effective. Note it will make more sense and will be cleaner when we're able to mark the number as a "lower bound": `0` is always a lower bound of the total hit count, so the method would actually be telling the truth.
2. Change the flow of information so that there are two different paths:
  * When loading the whole result (including the total hit count, call `ElasticsearchLoadableSearchResult#loadBlocking()` as we used to. (Same for Lucene)
  * When loading just hits, call a new method: call `ElasticsearchLoadableSearchResult#loadHitsBlocking()`, which only returns a `List<H>`. (Same for Lucene). It shouldn't be too hard since `loadBlocking` is already mostly just about loading hits; I'm sure you'll be able to factorize code easily.

Reply:
I fixed it before reading this comment using the solution (1). Anyway this is only temporary, since as you wrote we're going to change the API supporting the lower bound hit counts.